### PR TITLE
[Feat]: Like/Dislike

### PR DESCRIPTION
**Pull request by parthpandyappp**
- User is able to Like/Dislike a video.
- User can list all the liked videos at `/liked` route.

### CODE REVIEW

File: src/components/VideoCard.jsx [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+  addToWatchHistory,
+  removeFromLikes,
+} from "../helper-functions";
 import { useGenericData } from "../contexts";
 
 function VideoCard({ video }) {
   const { boolFunc } = useGenericData();
 
   return (
     <div className="card-main">
```
Comment (bharati-21) [inline src/components/VideoCard.jsx:13]: Suggestion - you can destructure video here before using the object keys.

File: src/components/VideoCard.jsx [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-      <Link to={`/video/${video._id}`}>
+      <Link
+        to={`/video/${video._id}`}
+        onClick={() => addToWatchHistory(video, boolFunc)}
+      >
         <img className="img-preview" src={video.imgPreview} alt="vid" />
```
Comment (bharati-21) [inline src/components/VideoCard.jsx:18]: Suggestion - you can give a proper alt attribute.

File: src/components/VideoCard.jsx [hunk 3]
```diff
@@ -28,6 +37,20 @@ function VideoCard({ video }) {
...
+        <div className="utils-opts">
+          <button
+            className="btn btn-primary"
+            onClick={() => removeFromLikes(video._id, boolFunc)}
+          >
+            Remove from Liked Videos
+          </button>
+        </div>
+      ) : (
+        ""
```
Comment (bharati-21) [inline src/components/VideoCard.jsx:52]: Instead of returning an empty string, you can return null.

File: src/helper-functions/like-helpers.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        const response = await axios({
+            method: "GET",
+            headers: {
+                authorization: encodedToken, // passing token as an authorization header
+            },
+            url: "/api/user/likes",
+        });
+        dispatch({ type: "SET_LIKED_DATA", payload: response.data.likes });
+    } catch (error) {
+        console.log(error);
```
Comment (bharati-21) [inline src/helper-functions/like-helpers.js:15]: Please remove log statements.

File: src/helper-functions/like-helpers.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            },
+            url: "/api/user/likes",
+            data: JSON.stringify({
+                video: videoObj,
+            }),
+        });
+
+        boolFunc();
+    } catch (error) {
+        console.log(error);
```
Comment (bharati-21) [inline src/helper-functions/like-helpers.js:38]: Here as well.

File: src/helper-functions/like-helpers.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            method: "DELETE",
+            headers: {
+                authorization: encodedToken,
+            },
+            url: `/api/user/likes/${id}`,
+        });
+
+        boolFunc();
+    } catch (error) {
+        console.log(error);
```
Comment (bharati-21) [inline src/helper-functions/like-helpers.js:56]: Here as well.

File: src/helper-functions/like-helpers.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        const encodedToken = localStorage.getItem("token");
+        await axios({
+            method: "DELETE",
+            headers: {
+                authorization: encodedToken,
+            },
+            url: `/api/user/likes/${id}`,
+        });
+
+        boolFunc();
```
Comment (bharati-21) [inline src/helper-functions/like-helpers.js:54]: Also, you could use a better name. I saw the same function name earlier. But it does not make sense to me what `boolFunc` does.

File: src/helper-functions/watchhistory-helpers.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        });
+
+        boolFunc();
+    } catch (error) {
+        console.log(error);
+    }
+};
+
+
+export { addToWatchHistory, removeFromHistory, removeAllFromHistory, getHistoryData }
```
Comment (bharati-21) [inline src/helper-functions/watchhistory-helpers.js:78]: Please remove log statements everywhere.

Comment (bharati-21) [reply src/helper-functions/watchhistory-helpers.js:78]: And kindly use a more readable function name.

File: src/pages/VideoDetail.jsx [hunk 9]
```diff
@@ -22,6 +27,14 @@ function VideoDetail() {
...
+    if (state.Liked.length > 0) {
+      console.log("once");
+      return state.Liked.find((item) => item._id === id);
+    }
+    return false;
+  };
+
   return (
     <div className="main-container">
       <SideNav />
```
Comment (bharati-21) [inline src/pages/VideoDetail.jsx:40]: Suggestion - I see you have used SideNav across multiple components. If they are used in all the private routes, you could maybe add this as a commonly shared component.

**Review by bharati-21 — APPROVED** [review]
Only 2 major suggestions - please remove logs and also use meaningful names for your functions.

Apart from that all looks good. Just dropped a few suggestions you could follow.

**Comment by parthpandyappp** [comment]
Thankyou @bharati-21 for putting in your valuable reviews here :+1: I'll look into it.

**Pull request merged**